mutiny-wasm/src/indexed_db.rs

```diff
@@ -61,6 +68,65 @@ impl IndexedDbStorage {
         })
     }
 
+    /// Read the mnemonic from indexed db, if one does not exist,
+    /// then generate a new one and save it to indexed db.
+    pub(crate) async fn get_mnemonic(
+        override_mnemonic: Option<Mnemonic>,
+        password: Option<&str>,
+        cipher: Option<Cipher>,
+    ) -> Result<Mnemonic, MutinyError> {
+        let indexed_db = Self::build_indexed_db_database().await?;
+        let tx = indexed_db
+            .transaction(&[WALLET_OBJECT_STORE_NAME], TransactionMode::ReadWrite)
+            .map_err(|e| {
+                MutinyError::read_err(
+                    anyhow!("Failed to create indexed db transaction: {e}").into(),
+                )
+            })?;
+
+        let store = tx.store(WALLET_OBJECT_STORE_NAME).map_err(|e| {
+            MutinyError::read_err(anyhow!("Failed to create indexed db store: {e}").into())
+        })?;
+
+        let key = JsValue::from(MNEMONIC_KEY);
+        let read = store
+            .get(&key)
+            .await
+            .map_err(|_| MutinyError::read_err(MutinyStorageError::IndexedDBError))?;
+
+        // if there is no mnemonic in indexed db generate a new one and insert
+        let res = if read.is_null() || read.is_undefined() {
+            let seed = override_mnemonic.unwrap_or_else(|| generate_seed(12).unwrap());
+
+            // encrypt and save to indexed db
+            let value = encrypt_value(MNEMONIC_KEY, serde_json::to_value(seed.clone())?, cipher)?;
+            store
+                .put(&JsValue::from_serde(&value)?, Some(&key))
+                .await
+                .map_err(|_| MutinyError::write_err(MutinyStorageError::IndexedDBError))?;
+
+            seed
+        } else {
+            // if there is a mnemonic in indexed db, then decrypt it
+            let value = decrypt_value(MNEMONIC_KEY, read.into_serde()?, password)?;
+
+            let seed: Mnemonic = serde_json::from_value(value)?;
+
+            // if we hae an override mnemonic, then we need to check that it matches the one in indexed db
+            if override_mnemonic.is_some_and(|m| m != seed) {
+                panic!("Mnemonic already exists in indexed db, cannot override");
+            }
```

Comment:
Why override mnemonic?

Reply:
We currently support entering a mnemonic, we just didnt handle if a wallet was already created. This will handle it better